### PR TITLE
content= argument for make_schema() and make_data()

### PR DESCRIPTION
**Pull request by simonw**
To allow you to pass in a string of YAML in place of a file path.

Refs https://github.com/23andMe/Yamale/issues/104

### CODE REVIEW

File: yamale/readers/tests/test_yaml.py [hunk 1]
```diff
@@ -9,14 +10,27 @@
...
+    if use_string:
+        with io.open(TYPES, encoding='utf-8') as f:
+            content = f.read()
+        a = yaml_reader.parse_yaml(parser=parser, content=content)[0]
+    else:
+        a = yaml_reader.parse_yaml(TYPES, parser)[0]
     assert a['string'] == 'str()'
 
 
+def test_parse_validates_arguments():
```
Comment (mildebrandt) [inline yamale/readers/tests/test_yaml.py:24]: Thanks for the negative test cases!

**Comment by mildebrandt** [comment]
I wonder if we can fix this error message:
```
yamale.validate(schema, data)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/chrism/workspace/simonw/Yamale/yamale/yamale.py", line 49, in validate
    raise YamaleError(results)
yamale.yamale_error.YamaleError: Error validating data 'None' with 'None'
        age: '26a' is not a int.
```

Since there are no filenames, we get `None` for the data and schema. I think we have three cases:
- No file for schema or data: `Error validating data`
- No file for schema: `Error validating data <filename>`
- No file for data: `Error validating data with schema <filename>`

I'm open to other phrasing too.

**Comment by simonw** [comment]
Good catch, I hadn't seen that. I like those three error messages.

**Comment by simonw** [comment]
OK, implemented those messages. Figuring out how to test it was a bit tricky but I'm happy with the solution I came up with: https://github.com/23andMe/Yamale/blob/ad676dbf9ad10bf7a827ba1c7027e90e10046f05/yamale/tests/test_functional.py#L325-L348

File: yamale/schema/validationresults.py [hunk 2]
```diff
@@ -26,8 +26,13 @@ def __init__(self, data, schema, errors):
...
-        else: 
-            head_line = "Error validating data '%s' with '%s'\n\t" % (self.data, self.schema)
+        else:
+            head_line_bits = ["Error validating data"]
+            if self.data:
+                head_line_bits.append("'{}'".format(self.data))
+            if self.schema:
+                head_line_bits.append("with schema '{}'".format(self.schema))
+            head_line = " ".join(head_line_bits)
             error_str = head_line + '\n\t'.join(self.errors)
```
Comment (mildebrandt) [inline yamale/schema/validationresults.py:36]: Thanks for the message formatting. Looks like the `\n\t` after the `head_line` was missed in refactoring.

**Comment by mildebrandt** [comment]
I think the rest looks great. We can merge this in after the last bit of message formatting. Thanks for contributing to Yamale!

**Comment by simonw** [comment]
Good catch, fixed.

File: yamale/schema/validationresults.py (hunk 2 above)
Comment (simonw) [reply yamale/schema/validationresults.py:37]: Fixed in https://github.com/23andMe/Yamale/pull/105/commits/23b49e6d9f807584441eab7ca8b26c867cc1a405#diff-cbb0754bd27ec59a69abd0f651e48c3b

**Pull request merged**